instrumentation/net/http/httptrace/otelhttptrace/clienttrace_test.go

```diff
@@ -32,3 +34,38 @@ func ExampleNewClientTrace() {
 
 	fmt.Println(resp.Status)
 }
+
+// TestNewClientParallelismWithoutSubspans tests running many Gets on a client simultaneously,
+// which would trigger a race condition if root were not protected by a mutex.
+func TestNewClientParallelismWithoutSubspans(t *testing.T) {
+	t.Parallel()
+
+	makeClientTrace := func (ctx context.Context) *httptrace.ClientTrace{
+		return NewClientTrace(ctx, WithoutSubSpans())
+	}
+
+	client := http.Client{
+		Transport: otelhttp.NewTransport(
+			http.DefaultTransport,
+			otelhttp.WithClientTrace(makeClientTrace),
+		),
+	}
+
+	var wg sync.WaitGroup
+
+	for i := 1; i < workers; i++ {
+		wg.Add(1)
+		go func() {
+			resp, err := client.Get("https://example.com")
+			if err != nil {
+				fmt.Println(err)
+				return
+			}
+			resp.Body.Close()
+			fmt.Println(resp.Status)
```

Comment:
This should either be an `assert`, or not be there (tests souldn't use `Println`).

Reply:
Done.